Task_level_Parallelism/Data_driven/unique_task_regions/test.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 1986-2022 Xilinx, Inc. All Rights Reserved.
+ * Copyright 2022-2023 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "test.h"
+
+void copy(hls::stream<int>& in, hls::stream<int>& out) {
+    for (int i = 0; i < 8; i++)
+        out.write(in.read());
+}
+
+void func(hls::stream<int>& in, hls::stream<int>& out) {
+    int cnt = 0;
+    // This loop is only needed for the persistent variable above
```

Comment:
why is this while loop needed here exactly?

Reply:
It's not pipelined due to possible complexities for the compiler to figure out the dependency (of cnt) btw possible  concurrent passes?
Or I got the understanding wrong?